packages/common/src/transports/wiresaw.ts

```diff
@@ -0,0 +1,168 @@
+import { EIP1193RequestFn, Hex, RpcTransactionReceipt, Transport } from "viem";
+import { estimateUserOperationGas } from "./methods/estimateUserOperationGas";
+import { getUserOperationReceipt } from "./methods/getUserOperationReceipt";
+
+type WiresawSendUserOperationResult = {
+  txHash: Hex;
+  userOpHash: Hex;
+};
+
+type WiresawOptions<transport extends Transport> = {
+  /** wiresaw-compatible transport */
+  wiresawTransport: transport;
+  /** fallback transport for bundler RPC methods */
+  fallbackBundlerTransport?: Transport;
+  /** fallback transport for all other RPC methods */
+  fallbackDefaultTransport?: Transport;
+};
+
+export function wiresaw<const wiresawTransport extends Transport>({
+  wiresawTransport,
+  fallbackBundlerTransport,
+  fallbackDefaultTransport,
+}: WiresawOptions<wiresawTransport>): wiresawTransport {
+  return ((opts) => {
```

Comment:
iirc we have access to the chain here, so couldn't we do all the transport handling here internally?

Reply:
wonder what happens if we throw in that case 🤣

Reply:
feels fine if we say "can only use wiresaw transport with a client that defines a chain"

or take in a chain param so wagmi transports look like

```
[garnet.id]: wiresaw({ chain: garnet })
```

Reply:
updated to allow `wiresaw()` as default or overriding the transports explicitly (similar to how other transports allow an explicit url to be passed in)